packages/cinterop/src/commonMain/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -225,23 +233,28 @@ expect object RealmInterop {
 
     fun realm_get_col_key(realm: RealmPointer, classKey: ClassKey, col: String): PropertyKey
 
-    fun realm_get_value(obj: RealmObjectPointer, key: PropertyKey): RealmValue
-    fun realm_set_value(obj: RealmObjectPointer, key: PropertyKey, value: RealmValue, isDefault: Boolean)
+    fun MemTrackingAllocator.realm_get_value(obj: RealmObjectPointer, key: PropertyKey): RealmValue?
+    fun realm_set_value(
+        obj: RealmObjectPointer,
+        key: PropertyKey,
+        value: RealmValue,
+        isDefault: Boolean
+    )
     fun realm_set_embedded(obj: RealmObjectPointer, key: PropertyKey): RealmObjectPointer
     fun realm_object_add_int(obj: RealmObjectPointer, key: PropertyKey, value: Long)
 
     // list
     fun realm_get_list(obj: RealmObjectPointer, key: PropertyKey): RealmListPointer
     fun realm_get_backlinks(obj: RealmObjectPointer, sourceClassKey: ClassKey, sourcePropertyKey: PropertyKey): RealmResultsPointer
     fun realm_list_size(list: RealmListPointer): Long
-    fun realm_list_get(list: RealmListPointer, index: Long): RealmValue
-    fun realm_list_add(list: RealmListPointer, index: Long, value: RealmValue)
+    fun MemTrackingAllocator.realm_list_get(list: RealmListPointer, index: Long): RealmValue?
+    fun RealmValue.realm_list_add(list: RealmListPointer, index: Long)
     fun realm_list_insert_embedded(list: RealmListPointer, index: Long): RealmObjectPointer
     // Returns the element previously at the specified position
-    fun realm_list_set(list: RealmListPointer, index: Long, value: RealmValue): RealmValue
+    fun RealmValue.realm_list_set(list: RealmListPointer, index: Long)
```

Comment:
Please disregard the `RealmValue` in front of the function.

I had to change the implementation to do the `set` here completely separate from the `get` (which is done at an SDK level) because the `realm_value_t` struct gets otherwise updated by core if I place everything inside the same scope. I tested it by doing something like this:
```
// outputTransport, value stored at index 1 = 1
// inputTransport, value to set at index 1 = 2
val outputTransport = realmc.realm_list_get(list, index, outputStruct) // outputTransport contains 1
realmc.realm_list_set(list.cptr(), index, inputTransport.value) // outputTransport changed to 2 without having modified it
```